public/app/plugins/datasource/cloudwatch/types.ts

```diff
@@ -144,46 +137,7 @@ export interface CloudWatchSecureJsonData extends AwsAuthDataSourceSecureJsonDat
   secretKey?: string;
 }
 
-export interface GetQueryResultsRequest {
-  /**
-   * The ID number of the query.
-   */
-  queryId: string;
-}
-
-export interface ResultField {
-  /**
-   * The log event field.
-   */
-  field?: string;
-  /**
-   * The value of this field.
-   */
-  value?: string;
-}
-
-export interface QueryStatistics {
-  /**
-   * The number of log events that matched the query string.
-   */
-  recordsMatched?: number;
-  /**
-   * The total number of log events scanned during the query.
-   */
-  recordsScanned?: number;
-  /**
-   * The total number of bytes in the log events scanned during the query.
-   */
-  bytesScanned?: number;
-}
-
-export type QueryStatus = 'Scheduled' | 'Running' | 'Complete' | 'Failed' | 'Cancelled' | string;
-
-export type CloudWatchLogsRequest =
-  | GetLogEventsRequest
-  | StartQueryRequest
-  | DescribeLogGroupsRequest
-  | GetLogGroupFieldsRequest;
+export type CloudWatchLogsRequest = GetLogEventsRequest | StartQueryRequest | QueryParam;
```

Comment:
I guess this is old but what's "GetLogEventsRequest"?

Reply:
The CloudWatch datasource file implement the [DataSourceWithLogsContextSupport](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/logs.ts#L154) interface. That means the platform (probably logs query runner or logs panel) will call the [getLogRowContext](https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/cloudwatch/datasource.ts#L137) method in the data source file. I'm not familiar with this method and I don't know when it's being triggered, but it seems like what CloudWatch logs does is that it will fetch log event from the `GetLogEvents` api, tweak the response to the format that DataSourceWithLogsContextSupport expects. 

Since this is not a data query, it should be using the call resource api instead. It's the last subType request that should be moved to the resource api instead. See [this epic](https://github.com/grafana/grafana/issues/57146) or more details.